operator/internal/webhook/admission/pcs/authorization/decoder.go

```
@@ -0,0 +1,127 @@
+)
+
+var (
+	pcsGVK            = grovecorev1alpha1.SchemeGroupVersion.WithKind(constants.KindPodCliqueSet)
+	pcsgGVK           = grovecorev1alpha1.SchemeGroupVersion.WithKind(constants.KindPodCliqueScalingGroup)
+	pclqGVK           = grovecorev1alpha1.SchemeGroupVersion.WithKind(constants.KindPodClique)
+	podGVK            = corev1.SchemeGroupVersion.WithKind("Pod")
+	secretGVK         = corev1.SchemeGroupVersion.WithKind("Secret")
+	roleGVK           = rbacv1.SchemeGroupVersion.WithKind("Role")
+	roleBindingGVK    = rbacv1.SchemeGroupVersion.WithKind("RoleBinding")
+	serviceGVK        = corev1.SchemeGroupVersion.WithKind("Service")
+	serviceAccountGVK = corev1.SchemeGroupVersion.WithKind("ServiceAccount")
+	hpaV2GVK          = autoscalingv2.SchemeGroupVersion.WithKind("HorizontalPodAutoscaler")
+	hpaV1GVK          = autoscalingv1.SchemeGroupVersion.WithKind("HorizontalPodAutoscaler")
+	podgangGVK        = groveschedulerv1alpha1.SchemeGroupVersion.WithKind("PodGang")
+
+	errDecodeRequestObject  = errors.New("failed to decode request")
+	errUnsupportedOperation = errors.New("unsupported operation")
+)
+
+// requestDecoder decodes the admission requests.
+// It optimizes only gets PartialObjectMetadata for the resource
+// as there is no need to get the full resource.
+type requestDecoder struct {
+	decoder admission.Decoder
+}
+
+func newRequestDecoder(mgr manager.Manager) *requestDecoder {
+	return &requestDecoder{
+		decoder: admission.NewDecoder(mgr.GetScheme()),
+	}
+}
+
+func (d *requestDecoder) decode(logger logr.Logger, req admission.Request) (*metav1.PartialObjectMetadata, error) {
+	reqGVK := schema.GroupVersionKind{
+		Group:   req.Kind.Group,
+		Version: req.Kind.Version,
+		Kind:    req.Kind.Kind,
+	}
+	switch reqGVK {
+	case pcsgGVK, pclqGVK, podGVK, serviceAccountGVK, serviceGVK, secretGVK, roleGVK, roleBindingGVK,
+		hpaV2GVK, hpaV1GVK,
+		podgangGVK:
+		return d.decodeAsPartialObjectMetadata(req)
+	default:
```

You will also get requests for `Scale` subresource. That will go to the default case where it will print `unknown GVK` but that is misleading because you know of the `Scale` subresource but we choose to ignore it, right?